File: prime_simple.go
package main

func main() {
	const N = 20000000

	count := 0
	for i := 2; i < N; i++ {
		if isPrime(i) {
			count += 1
		}
	}

	println(count)
}

func isPrime(value int) bool {
	for i := 2; i*i <= value; i++ {
		if value%i == 0 {
			return false
		}
	}
	return true
}

//Compiled: go build -ldflags="-s -w"
//Run Time: 34.5 sec
//Memory: 1,490 K

File: prime_simple.rs
fn main() {
  let count = (2..10000000).filter(|&p| is_prime(p)).count();
  println!("{}", count);
}

fn is_prime(n: u64) -> bool {
  (2..).take_while(|x| x * x <= n).all(|i| n % i != 0)
}

//Compiled: cargo build --release
//Run Time: 11.9 sec
//Memory: 548 K
